src/app/customizable-demo-form.component.ts

```diff
@@ -164,6 +173,73 @@ export class CustomizableDemoForm {
   // Whether to hide the loading icon when the score is below the minimum
   // threshold to show feedback.
   hideLoadingIconForScoresBelowMinThreshold = false;
+  // Whether to always hide the loading icon.
+  alwaysHideLoadingIcon = false;
+
+  demoSettings: DemoSettings|null = null;
+
+  constructor(private route: ActivatedRoute, private router: Router) {}
+
+  ngOnInit() {
+    this.route.params.subscribe((params: Params) => {
+      if (params['uiSettings']) {
+        const decodedUISettings: UISettings = JSON.parse(
+              decodeURIComponent(params['uiSettings'] as string));
+        this.useCustomColorScheme = decodedUISettings.useCustomColorScheme;
+        this.useCustomFeedbackText = decodedUISettings.useCustomFeedbackText;
+        this.customizeScoreThresholds = decodedUISettings.customizeScoreThresholds;
+      }
+      if (params['encodedDemoSettings']) {
+        const decodedDemoSettings: DemoSettings = JSON.parse(
+            decodeURIComponent(params['encodedDemoSettings'] as string));
+        console.debug('I see demo settings in the url:', decodedDemoSettings);
+        this.configuration = decodedDemoSettings.configuration;
+        if (this.useCustomColorScheme) {
+          this.customColorScheme = decodedDemoSettings.gradientColors;
+        } else {
+          for (let i = 0; i < this.colorSchemes.length; i++) {
+            let colorScheme = this.colorSchemes[i];
+            if (colorScheme.colors.every(
+                (element, index) => element === decodedDemoSettings.gradientColors[index])) {
+              this.selectedColorScheme = this.colorSchemes[i];
```

Comment:
I can't quite figure out what this is doing...can you either add a comment or move it to a function with a descriptive name?  Same for below loop.

Reply:
Done (commit not pushed yet, addressing each comment one at a time)